### PR TITLE
Add a linter via GH workflows

### PR DESCRIPTION
**Pull request by pnovotnak**
https://github.com/golangci/golangci-lint-action

Gofmt was performed using this command: `gofmt ./admiral/...`

### CODE REVIEW

**Comment by aattuluri** [comment]
Thanks for the PR, wanted to do this for so long, there seems to be merge conflict

**Comment by aattuluri** [comment]
@pnovotnak sorry, there are conflicts looks like

**Comment by aattuluri** [comment]
@pnovotnak Waiting for the conflict resolution to merge this.

File: install/admiral/overlays/demosinglecluster/envconfig_values.yaml [hunk 1]
```diff
@@ -19,4 +19,4 @@ spec:
             - admiral-sync
             - --sync_period
             - 20s
-          name: admiral
\ No newline at end of file
+          name: admiraladmiral/pkg/controller/admiral/service_test.go
```
Comment (aattuluri) [inline install/admiral/overlays/demosinglecluster/envconfig_values.yaml:22]: An accidental change?

**Comment by aattuluri** [comment]
@pnovotnak The build failed, probably dependency on some tool not checked in?

File: install/admiral/overlays/demosinglecluster/envconfig_values.yaml (hunk 1 above)
Comment (pnovotnak) [reply install/admiral/overlays/demosinglecluster/envconfig_values.yaml:22]: Thanks for the careful review. Yes, that was accidental

**Comment by pnovotnak** [comment]
I'm not sure what to do about this test

Screen Shot 2020-06-22 at 2 46 11 PM
Screen Shot 2020-06-22 at 2 46 16 PM

File: admiral/pkg/clusters/serviceentry_test.go [hunk 2]
```diff
@@ -382,7 +382,7 @@ func TestModifyExistingSidecarForLocalClusterCommunication(t *testing.T) {
 			t.Fail()
 		}
 
-		hostList := append(createdSidecar.Spec.Egress[0].Hosts, "test-dependency-namespace/test-local-fqdn", "test-dependency-namespace/test.myservice.global")
```
[the remark this answers is not in the document]
Comment (pnovotnak) [reply admiral/pkg/clusters/serviceentry_test.go:385]: This was an attempt to fix the test. See above comment

[the remark this answers is not in the document]
Comment (aattuluri) [reply admiral/pkg/clusters/serviceentry_test.go:385]: @pnovotnak On my branch, if I use `hostList := append(createdSidecar.Spec.Egress[0].Hosts, "test-dependency-namespace/test-local-fqdn", "test-dependency-namespace/test.myservice.global")` the test passes, if I swap the order and use `hostList := append(createdSidecar.Spec.Egress[0].Hosts, "test-dependency-namespace/test.myservice.global", "test-dependency-namespace/test-local-fqdn")` which is what you have in the PR, then the test fails. Are you seeing this test fail no matter what?

Either ways, do we need the above line change for linter check to pass? if it is then I don't understand why linter cares about elements order, ideally it shouldn't.

**Comment by pnovotnak** [comment]
The test fails either way

The linter does not care about order

**Comment by aattuluri** [comment]
@pnovotnak Ok now its the linter errors, I see that the tests passed, hope you are taking a look?. 
Also, I think we need to use something like `assert.ElementsMatch(t, expectedSlice, actualSlice)` (https://github.com/stretchr/testify/issues/275) for the comparison in that test case.

**Comment by pnovotnak** [comment]
Yes, taking a look. Lots of changes but they should be quick to resolve, assuming the linter isn't producing false positives.

> I think we need to use something like assert.ElementsMatch(t, expectedSlice, actualSlice) (stretchr/testify#275) for the comparison in that test case.

I agree, ordering shouldn't cause test to flake in this case. Going to leave that for since we know it's a transient issue

**Comment by aattuluri** [comment]
@pnovotnak Still working on this I believe?

**Comment by pnovotnak** [comment]
The changes to test are kinda horrifying but they work

https://github.com/istio-ecosystem/admiral/compare/b2ef66c...5e9219d#diff-1a964207b5a08ac3bb3d631495b4ce7eL385-R420

**Pull request merged**